src/ngx_lua_resty_lmdb_module.c

```diff
@@ -289,26 +282,108 @@ static ngx_int_t ngx_lua_resty_lmdb_init_worker(ngx_cycle_t *cycle)
         block_size = EVP_CIPHER_block_size(cipher);
         if (block_size == 0) {
             ngx_log_error(NGX_LOG_CRIT, cycle->log, 0,
-                "unable to set LMDB encryption key, block size");
+                          "unable to set LMDB encryption key, block size");
             return NGX_ERROR;
         }
 
         rc = mdb_env_set_encrypt(lcf->env, lmdb_encrypt_func, &enckey, block_size);
         if (rc != 0) {
             ngx_log_error(NGX_LOG_CRIT, cycle->log, 0,
-                "unable to set LMDB encryption key: %s", mdb_strerror(rc));
+                          "unable to set LMDB encryption key: %s",
+                          mdb_strerror(rc));
             return NGX_ERROR;
         }
 
-        /* destroy data*/
-        ngx_explicit_memzero(lcf->key_data.data, lcf->key_data.len);
-        ngx_explicit_memzero(lcf->encryption_mode.data, lcf->encryption_mode.len);
+        /* worker will destroy secret data */
+        if (is_master == 0) {
```

Comment:
```suggestion
        if (is_master) {
```

Reply:
It means the Nginx worker will destroy the secret data, so the condition should be `is_master == 0` or `!is_master`.

Reply:
sorry, it's my typo.